src/helpers.coffee

```diff
@@ -268,6 +268,20 @@ exports.nameWhitespaceCharacter = (string) ->
     when '\t' then 'tab'
     else string
 
+exports.parseNumber = (string) ->
+  return NaN unless string?
+
+  base = switch string.charAt 1
+    when 'b' then 2
+    when 'o' then 8
+    when 'x' then 16
+    else null
+
+  if base?
+    parseInt string[2..].replace(/_/g, ''), base
```

Comment:
`parseInt()` doesn't understand numeric separators (`_`) apparently, so strip them out first

Otherwise the raw `value` in the AST was incorrect for eg `10_000`